src/Components/Server/src/DependencyInjection/RazorComponentsServiceCollectionExtensions.cs

```diff
@@ -1,115 +1,57 @@
+using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Components.Server;
 using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.Components.Server.DependencyInjection;
 using Microsoft.AspNetCore.Components.Services;
 using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Microsoft.JSInterop;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
     /// <summary>
-    /// Extension methods to configure an <see cref="IServiceCollection"/> for interactive components.
+    /// Extension methods to configure an <see cref="IServiceCollection"/> for components.
     /// </summary>
     public static class RazorComponentsServiceCollectionExtensions
     {
-        /// <summary>
-        /// Adds Razor Component services to the service collection.
-        /// </summary>
-        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
-        /// <param name="startupType">A Razor Components project startup type.</param>
-        /// <returns>The <see cref="IServiceCollection"/>.</returns>
-        public static IServiceCollection AddRazorComponents(
-            this IServiceCollection services,
-            Type startupType)
-        {
-            if (services == null)
-            {
-                throw new ArgumentNullException(nameof(services));
-            }
-
-            if (startupType == null)
-            {
-                throw new ArgumentNullException(nameof(startupType));
-            }
-
-            return AddRazorComponentsCore(services, startupType);
-        }
-
         /// <summary>
         /// Adds Razor Component app services to the service collection.
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection"/>.</param>
-        /// <typeparam name="TStartup">A Components app startup type.</typeparam>
         /// <returns>The <see cref="IServiceCollection"/>.</returns>
-        public static IServiceCollection AddRazorComponents<TStartup>(
-            this IServiceCollection services)
+        public static IServiceCollection AddRazorComponents(this IServiceCollection services)
         {
-            if (services == null)
-            {
-                throw new ArgumentNullException(nameof(services));
-            }
-
-            return AddRazorComponentsCore(services, typeof(TStartup));
-        }
+            // We've discussed with the SignalR team and believe it's OK to have repeated
+            // calls to AddSignalR (making the nonfirst ones no-ops). If we want to change
+            // this in the future, we could change AddComponents to be an extension
+            // method on ISignalRServerBuilder so the developer always has to chain it onto
+            // their own AddSignalR call. For now we're keeping it like this because it's
+            // simpler for developers in common cases.
+            services.AddSignalR().AddMessagePackProtocol();
```

Comment:
IMO this is already the obvious thing to do. I dunno why we would consider everything else 😆

Reply:
Oh, I see it's a copy-paste. I'm not sure this comment is useful .

Reply:
Should this be AddComponents instead of AddRazorComponents?

Reply:
I killed the comments